pkg/jirautils/jirautil.go

```diff
@@ -0,0 +1,132 @@
+package jirautils
+
+import (
+	"bytes"
+
+	jira "github.com/andygrunwald/go-jira"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	client                     *jira.Client
+	isJiraConnectionSuccessful bool
+
+	//AccountID for bug assignment
+	AccountID string
+)
+
+const (
+	jiraURL = "https://portworx.atlassian.net/"
+)
+
+// Init function for the Jira
+func Init(username, token string) {
+	httpClient := jira.BasicAuthTransport{
+		Username: username,
+		Password: token,
+	}
+	var err error
+
+	client, err = jira.NewClient(httpClient.Client(), jiraURL)
+	if err != nil {
+		isJiraConnectionSuccessful = false
+		logrus.Errorf("Testrail connection not successful, Cause: %v", err)
+
+	}
+
+	isJiraConnectionSuccessful = true
+
+}
+
+// CreateIssue creates issue in jira
+func CreateIssue(issueDesription, issueSummary string) string {
+
+	issueKey := ""
+
+	if isJiraConnectionSuccessful {
+
+		issueKey = createPTX(issueDesription, issueSummary)
+	} else {
+		logrus.Warn("Skipping issue creation as jira connection is not successful")
+	}
+	return issueKey
+
+}
+
+func getPTX(issueID string) {
+
+	issue, _, err := client.Issue.Get(issueID, nil)
+	logrus.Infof("Error: %v", err)
+
+	logrus.Infof("%s: %+v\n", issue.Key, issue.Fields.Summary)
+	logrus.Infof("%+v\n", issue)
+
+	logrus.Infof("%s: %s\n", issue.ID, issue.Fields.Summary)
+	logrus.Info(issue.Fields.FixVersions[0].Name)
+
+}
+
+func createPTX(description, summary string) string {
+
+	i := jira.Issue{
+		Fields: &jira.IssueFields{
+			Assignee: &jira.User{
+				AccountID: AccountID,
+			},
+			Description: description,
+			Type: jira.IssueType{
+				Name: "Bug",
+			},
+			Project: jira.Project{
+				Key: "PTX",
+			},
+			FixVersions: []*jira.FixVersion{
+				{
+					Name: "master",
+				},
+			},
+			AffectsVersions: []*jira.AffectsVersion{
+				{
+					Name: "master",
+				},
+			},
+			Summary: summary,
+		},
+	}
+	issue, resp, err := client.Issue.Create(&i)
+
+	logrus.Infof("Resp: %v", resp.StatusCode)
+	issueKey := ""
+	if resp.StatusCode == 201 {
+		logrus.Info("Successfully created new jira issue.")
+		logrus.Infof("Jira Issue: %+v\n", issue.Key)
+		issueKey = issue.Key
+
+	} else {
+		logrus.Infof("Error while creating jira issue: %v", err)
+		buf := new(bytes.Buffer)
+		buf.ReadFrom(resp.Body)
+		newStr := buf.String()
+		logrus.Infof(newStr)
+
+	}
+	return issueKey
+
+}
+
+func getProjects() {
```

Comment:
Are we using this function anywhere?

Reply:
not yet.. I might need this for my next changes, I will delete it in my next PR if not needed